sbi/utils/user_input_checks_utils.py

```diff
@@ -92,16 +105,26 @@ def __init__(
         return_type: Optional[torch.dtype] = float32,
         batch_shape=torch.Size(),
         event_shape=torch.Size(),
-        validate_args=None,
+        validate_args=False,
+        lower_bound: Optional[Tensor] = None,
+        upper_bound: Optional[Tensor] = None,
     ):
         super().__init__(
             batch_shape=batch_shape,
             event_shape=event_shape,
             validate_args=validate_args,
         )
 
+        if not validate_args:
+            warnings.warn(
+                """Validate args is set to false. If there are constraints on
+                prior parameters (e.g., >0), consider passing arg constraints as
+                documented in pytorch.distribution.Distribution"""
```

Comment:
Why is all of this run at `_set_mean_and_variance()`? Is it not enough to run it at `__init__()`?